Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,245 @@
+define([
+        '../Core/Cartesian3',
+        '../Core/Cartesian4',
+        '../Core/Check',
+        '../Core/Color',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/Intersect',
+        '../Core/Matrix4',
+        '../Core/Plane'
+    ], function(
+        Cartesian3,
+        Cartesian4,
+        Check,
+        Color,
+        defaultValue,
+        defined,
+        defineProperties,
+        Intersect,
+        Matrix4,
+        Plane) {
+    'use strict';
+
+    /**
+     * Specifies a set of clipping planes. Clipping planes selectively disable rendering in a region on the outside of the specified list of {@link Plane} objects.
+     *
+     * @alias ClippingPlanesCollection
+     * @constructor
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Plane[]} [options.planes=[]] An array of up to 6 {@link Plane} objects used to selectively disable rendering on the outside of each plane.
+     * @param {Boolean} [options.enabled=true] Determines whether the clipping planes are active.
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
+     * @param {Boolean} [options.combineClippingRegions=true] If true, the region to be clipped must be included in all planes in this collection. Otherwise, a region will be clipped if included in any plane in the collection.
+     * @param {Color} [options.edgeColor=Color.WHITE] The color applied to highlight the edge along which an object is clipped.
+     * @param {Number} [options.edgeWidth=0.0] The width, in pixels, of the highlight applied to the edge along which an object is clipped.
+     */
+    function ClippingPlanesCollection(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * An array of up to 6 {@link Plane} objects used to selectively disable rendering on the outside of each plane.
+         *
+         * @type {Plane}
+         * @default []
+         */
+        this.planes = defaultValue(options.planes, []);
+
+        /**
+         * Determines whether the clipping planes are active.
+         *
+         * @type {Boolean}
+         * @default true
+         */
+        this.enabled = defaultValue(options.enabled, true);
+
+        /**
+         * The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
+         *
+         * @type {Matrix4}
+         * @default Matrix4.IDENTITY
+         */
+        this.modelMatrix = defaultValue(options.modelMatrix, Matrix4.clone(Matrix4.IDENTITY));
```

Comment:
This should be `Matrix4.clone(defaultValue(options.modelMatrix, Matrix4.IDENTITY))` instead.  Or do it in an if statement.  Using `defaultValue` makes the `Matrxi4.clone` resolve first so we're creating a `new Matrix4` every time regardless of whether or not we're using it.

Reply:
Same comment for `edgeColor` below